dune_api_scripts/duneanalytics.py

```diff
@@ -0,0 +1,210 @@
+            print("New query has been posted with response:")
+            print(data)
+        else:
+            print(response.text)
+
+    def execute_query(self, query_id):
+        query_data = {"operationName": "ExecuteQuery", "variables": {"query_id": query_id, "parameters": [
+        ]}, "query": "mutation ExecuteQuery($query_id: Int!, $parameters: [Parameter!]!) {\n  execute_query(query_id: $query_id, parameters: $parameters) {\n    job_id\n    __typename\n  }\n}\n"}
+
+        self.session.headers.update({'authorization': f'Bearer {self.token}'})
+        response = self.session.post(GRAPH_URL, json=query_data)
+        if response.status_code == 200:
+            data = response.json()
+            print("query executed successfully with response:")
+            print(data)
+        else:
+            print(response.text)
+
+    def query_result_id(self, query_id):
+        """
+        Fetch the query result id for a query
+        :param query_id: provide the query_id
+        :return:
+        """
+        query_data = {"operationName": "GetResult", "variables": {"query_id": query_id},
+                      "query": "query GetResult($query_id: Int!, $parameters: [Parameter!]) "
+                               "{\n  get_result(query_id: $query_id, parameters: $parameters) "
+                               "{\n    job_id\n    result_id\n    __typename\n  }\n}\n"
+                      }
+
+        self.session.headers.update({'authorization': f'Bearer {self.token}'})
+
+        response = self.session.post(GRAPH_URL, json=query_data)
+        if response.status_code == 200:
+            data = response.json()
+            if 'errors' in data:
+                return None
+            result_id = data.get('data').get('get_result').get('result_id')
+            return result_id
+        else:
+            print(response.text)
+            return None
+
+    def query_result(self, result_id):
+        """
+        Fetch the result for a query
+        :param result_id: result id of the query
+        :return:
+        """
+        query_data = {"operationName": "FindResultDataByResult",
+                      "variables": {"result_id": result_id},
+                      "query": "query FindResultDataByResult($result_id: uuid!) "
+                               "{\n  query_results(where: {id: {_eq: $result_id}}) "
+                               "{\n    id\n    job_id\n    error\n    runtime\n    generated_at\n    columns\n    __typename\n  }"
+                               "\n  get_result_by_result_id(args: {want_result_id: $result_id}) {\n    data\n    __typename\n  }\n}\n"
+                      }
+
+        self.session.headers.update({'authorization': f'Bearer {self.token}'})
+
+        response = self.session.post(GRAPH_URL, json=query_data)
+        if response.status_code == 200:
+            data = response.json()
+            return data
+        else:
+            print(response.text)
+            return {}
```

Comment:
Weird that we are returning something without specifying that the function returns anything.

Not necessary in this PR, but at some point (if this code is going to be maintained), it would be worth declaring and introducing static type analysis.